app/src/main/java/com/example/meetup/SignUpActivity.java

```diff
@@ -0,0 +1,101 @@
+package com.example.meetup;
+
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.app.ProgressDialog;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.material.textfield.TextInputEditText;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+import com.parse.SignUpCallback;
+
+public class SignUpActivity extends AppCompatActivity {
+
+    private ImageView back;
+    private Button signUp;
+    private TextInputEditText username;
+    private TextInputEditText password;
+    private TextInputEditText passwordAgain;
+    private ProgressDialog progressDialog;
+    private GoogleSignInAccount account;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signup);
+
+        progressDialog = new ProgressDialog(SignUpActivity.this);
+
+        back = findViewById(R.id.back);
+        signUp = findViewById(R.id.signup);
+        username = findViewById(R.id.username);
+        password = findViewById(R.id.password);
+        passwordAgain = findViewById(R.id.passwordagain);
+
+        Intent intent = getIntent();
+        if (intent != null) {
+            account = intent.getParcelableExtra("account");
+            if (account != null)
+                username.setText(account.getDisplayName());
+        }
```

Comment:
We may need to wrap it with an if statement to check whether `savedInstanceState` is null like
```
if (savedInstanceState == null) {
    Intent intent = getIntent();
    if (intent != null) {
        account = intent.getParcelableExtra("account");
        if (account != null)
            username.setText(account.getDisplayName());
    }
}
```
At the same time, we need to store the field values by overriding [`onSaveInstanceState`](https://developer.android.com/reference/android/app/Activity#onSaveInstanceState(android.os.Bundle,%20android.os.PersistableBundle)). Otherwise, if the user switch out from the app and the switch back later, it may lost the filled information.